packages/amplication-server/src/core/subscription/subscription.controller.ts

```diff
@@ -39,6 +45,19 @@ export class SubscriptionController {
           updateStatusDto.id,
           createSubscriptionInput
         );
+        const userId = updateStatusDto.metadata?.userId;
+        if (
+          createSubscriptionInput.plan === EnumSubscriptionPlan.Pro &&
```

Comment:
I think this event should be triggered with every subscription creation, with the relevant metadata - for now, just "Pro" will work, but we will need to track other future plans in the same way. 
You should probably exclude "free"

Reply:
I thought that the point here is to register purchase that client completed by himself. Which means that "Enterprise" plan should not be registgred.

Reply:
Synced with Erez, event should be issued also when workspace was upgraded to "Enterprise" plan